app/browser/api/ledger.js

```diff
@@ -1114,6 +1114,10 @@ const enable = (state, paymentsEnabled) => {
     }, 15 * ledgerUtil.milliseconds.second)
   } else if (paymentsEnabled) {
     // on toggle
+    if (!togglePromotionTimeoutId) {
```

Comment:
Is this correct? shouldn't it be clearing when it's non-zero?

Reply:
no it's not. Fixed